PEP 8 Compliance, Elimination of Redundancies, CCEA re-write.

zerbeln opened the pull request:
Changes are mostly re-names to make code PEP 8 compliant and eliminating redundant and obsolete code.

The biggest change in functions is with the ccea.pyx. Neural network functions have been moved to a separate file in a separate class (neural_network), and the Ccea is a distinct class for evolving populations of weights for the neural network.

zerbeln committed:
Removed depracated files, and renamed the successor filess.
Altering naming convention for functions and parameters from longName to long_name.

 Changes to be committed:
	deleted:    code/agent_domain.py
	new file:   code/agent_domain.pyx
	deleted:    code/agent_domain_2.pyx
	deleted:    code/ccea.py
	renamed:    code/ccea_2.pyx -> code/ccea.pyx
	deleted:    code/reward.py
	new file:   code/reward.pyx
	deleted:    code/reward_2.pyx
	modified:   code/setup.py
	modified:   core.py
	modified:   gym_example.py
	modified:   mods.py
	modified:   rover_domain_core_gym.py
	modified:   run.py

zerbeln committed:
Naming conventions changed to long_name
Changes to be committed:
	renamed:    code/assignRandomPolicy.py -> code/assign_random_policy.py
	modified:   code/ccea.pyx
	modified:   gym_example.py
	modified:   rover_domain_core_gym.py
	modified:   run.py
	modified:   specifics.py

zerbeln committed:
Eliminating redundancies in data library. Specifics.py is now obsolete.
Core.py will soon be obsolete. Consolidated test parameters into parameters.py
and most of simulation_core has been moved to rover_domain_core_gym. Naming
conventions are still being worked on.
 Changes to be committed:
	modified:   code/ccea.pyx
	modified:   code/reward_history.py
	modified:   code/trajectory_history.py
	modified:   core.py
	modified:   gym_example.py
	modified:   mods.py
	new file:   parameters.py
	modified:   rover_domain_core_gym.py
	modified:   specifics.py

zerbeln committed:
Deleted obsolete files, fixed more naming conventions. Starting to add
framework for supervisor in D++.
Changes to be committed:
	modified:   code/reward.pyx
	modified:   code/world_setup.py
	deleted:    core.py
	modified:   gym_example.py
	modified:   mods.py
	modified:   rover_domain_core_gym.py
	deleted:    run.py
	deleted:    specifics.py
	new file:   supervisor.py

zerbeln committed:
Added ReadMe.txt which contains descriptions of functions in rover_domain_core.

zerbeln committed:
All naming conventions should now be PEP 8 compliant (may be a few stragglers)
world_setup.py functions were moved into rover_domain_core_gym.py and
world_setup.py was removed.

 Changes to be committed:
	modified:   code/reward.pyx
	modified:   code/reward_history.py
	modified:   code/save_to_pickle.py
	modified:   code/trajectory_history.py
	deleted:    code/world_setup.py
	modified:   copy_files.py
	modified:   gym_example.py
	modified:   rover_domain_core_gym.py
	modified:   supervisor.py

zerbeln committed:
Fixed reward function to be consistent with global evaluation for tightly
coupled systems. More naming fixes to long_name from longName.
Changes to be committed:
	modified:   code/ccea.pyx
	modified:   code/reward.pyx
	modified:   code/trajectory_history.py
	modified:   gym_example.py
	modified:   parameters.py
	modified:   rover_domain_core_gym.py

zerbeln committed:
CCEA.pyx being re-written. Code for neural network has been separated from
ccea.pyx into neural_network.pyx. CCEA and neural network are not currently
connected to the rest of the code base.
 Changes to be committed:
	modified:   code/assign_random_policy.py
	modified:   code/ccea.pyx
	modified:   code/curriculum.py
	new file:   code/neural_network.pyx
	modified:   code/reward.pyx
	modified:   gym_example.py
	modified:   parameters.py

zerbeln committed:
Neural network and ccea are now functioning properly. They are still not
connected with the rest of the code base. Major changes need to be made to
match outputs from neural net and ccea with the pre-existing functions in
rover domain.
 Changes to be committed:
	renamed:    code/agent_domain.pyx -> code/agent.pyx
	modified:   code/ccea.pyx
	modified:   code/neural_network.pyx
	modified:   code/setup.py
	modified:   copy_files.py
	modified:   gym_example.py
	modified:   mods.py
	modified:   parameters.py
	modified:   rover_domain_core_gym.py
	modified:   supervisor.py
	new file:   tester.py

zerbeln committed:
Neural network and CCEA are now connected with the rest of the code base.
gym_example.py now runs for training mode. There are still a number of code
base elements which need work: test mode needs to be worked on as do the
save files.
 Changes to be committed:
	modified:   code/agent.pyx
	modified:   code/ccea.pyx
	modified:   code/neural_network.pyx
	modified:   code/reward_history.py
	modified:   code/trajectory_history.py
	modified:   gym_example.py
	modified:   mods.py
	modified:   parameters.py
	modified:   rover_domain_core_gym.py

zerbeln committed:
data dictionary has been relocated to rover_domain_core_gym.py. Parameters
class now only contains those parameters which are constants.

 Changes to be committed:
	modified:   code/agent.pyx
	modified:   code/ccea.pyx
	modified:   code/neural_network.pyx
	modified:   code/reward.pyx
	modified:   code/trajectory_history.py
	modified:   gym_example.py
	modified:   parameters.py
	modified:   rover_domain_core_gym.py
	modified:   tester.py

zerbeln committed:
Only superficial changes made to a few files.
 Changes to be committed:
	modified:   gym_example.py
	modified:   parameters.py
	modified:   rover_domain_core_gym.py
	modified:   supervisor.py

zerbeln committed:
Reward history and trajectory history save-to-file functions execute without
error. Further modifications may need to be made depending on how well data
is captured. Rover domain needs to be tested to make sure it functions
correctly.
 Changes to be committed:
	modified:   code/reward_history.py
	modified:   code/trajectory_history.py
	modified:   gym_example.py
	modified:   mods.py
	modified:   parameters.py
	modified:   rover_domain_core_gym.py

zerbeln committed:
Reward history save function re-written so that each stat run is saved to an independent file
Some obsolete files removed. Reward function distances no longer squared.

 Changes to be committed:
	deleted:    code/assign_random_policy.py
	modified:   code/reward.pyx
	modified:   code/reward_history.py
	modified:   gym_example.py
	modified:   parameters.py
	modified:   rover_domain_core_gym.py
	deleted:    supervisor.py

zerbeln committed:
Fixed missing inequality in rover domain core

zerbeln committed:
Fixed issues with CCEA fitness assignment. Fixed issue with reward
calculations. Further tests need to be run to test domain functionality
 Changes to be committed:
	modified:   code/agent.pyx
	modified:   code/ccea.pyx
	modified:   code/neural_network.pyx
	modified:   code/reward.pyx
	modified:   code/trajectory_history.py
	modified:   gym_example.py
	modified:   parameters.py
	modified:   rover_domain_core_gym.py

zerbeln committed:
Fixed output files so that reward history is written to a single file instead
of a file for each stat run. Re-ordered execution of gym_example.py, each mod
is run for n stat runs instead of running all mods each stat run. CCEA still
needs some work. The rovers do not appear to be learning better policies.

 Changes to be committed:
	modified:   code/reward_history.py
	modified:   code/save_to_pickle.py
	modified:   gym_example.py
	modified:   mods.py
	modified:   parameters.py
	modified:   rover_domain_core_gym.py

zerbeln committed:
Found issue with CCEA. Offspring pop was being created twice. Issue has
been fixed. Offspring pop is now only created during down-select.
 Changes to be committed:
	modified:   code/ccea.pyx

zerbeln committed:
Changes rover_domain_core_gym.py from slots to modules per discussion held
by group after meeting.
 Changes to be committed:
	deleted:    ReadMe.txt
	modified:   code/agent.pyx
	modified:   code/ccea.pyx
	modified:   code/neural_network.pyx
	modified:   code/reward.pyx
	modified:   code/trajectory_history.py
	modified:   gym_example.py
	modified:   parameters.py
	modified:   rover_domain_core_gym.py
	deleted:    tester.py

Sir-Batman commented on code/agent.pyx:
```diff
@@ -0,0 +1,154 @@
+            agent_orientations[agent_id, 0] = dx/norm
+            agent_orientations[agent_id, 1] = dy/norm
+
+        # # Check if action moves agent within the world bounds
```
Why is this section commented out?

Sir-Batman commented on parameters.py:
```diff
@@ -0,0 +1,31 @@
+"""
+
+class Parameters:
+
```
Add a class description with short description of purpose, and some suggested parameters to set in this class object.

Sir-Batman commented on rover_domain_core_gym.py:
```diff
@@ -1,244 +1,141 @@
+from code.reward import calc_global_reward
+
+# World Setup ----------------------------------------------------------------------------------------
+# Randomly initalize agent positions on map
```
Change function-level comments to doc strings

Sir-Batman reviewed (changes requested):
# Major Comments:

- Remove `.idea/*` files from the commit and add a line to the `.gitignore` to explicitly ignore `.idea` directory

## ReST style doc strings needed in functions in:

-  `/code/agent.pyx`
- `/code/ccea.pyx`
- `/code/neural_network.pyx`
- `/code/reward_history.py`
- `/code/save_to_pickle.py`
- `/code/trajectory_history.py`
- `/rover_domain_core_gym.py`

Sir-Batman commented:
[`curriculum.py`](https://github.com/AADILab/Rover-Domain/pull/9/files#diff-fb2ea9e2dc70565a26bdaf85ac52774d) is completely commented out, what is the reasoning for this?

zerbeln committed:
Fixed some issues with domain mechanics, removed orientations for rovers

zerbeln committed:
Changed NN from sigmoid to tanh

zerbeln committed:
Rovers are not learning

zerbeln committed:
World reset was being called at the incorrect time. Domain appears functional now

zerbeln committed:
Distances were not being clipped in rewards. Reinstatetd clipping.

zerbeln committed:
Working on implementation of supervisor which suggests counterfactual
partners in D++.
 Changes to be committed:
	modified:   code/reward.pyx
	modified:   gym_example.py
	modified:   mods.py
	modified:   parameters.py

zerbeln committed:
Changed suggested counterfactual to closest rovers

zerbeln committed:
added comments

zerbeln committed:
Completely removed code for orientations

zerbeln committed:
supervisor now a separate file

zerbeln committed:
Supervisor using wrong distances, corrected

zerbeln committed:
Agents and POIs can now be statically or randomly initialized

zerbeln committed:
Added new counterfactual types in supervisor

zerbeln committed:
Changed variable typing in supervisor

zerbeln committed:
Branch is being revised to be consistent with new AADI rover domain base.
 Changes to be committed:
	new file:   README.md
	new file:   ccea.pyx
	deleted:    copy_files.py
	deleted:    folder_to_data.py
	deleted:    gym_example.py
	new file:   main.py
	new file:   neural_network.pyx
	modified:   parameters.py
	new file:   reward.pyx
	new file:   rover_domain.pyx
	deleted:    rover_domain_core_gym.py
	new file:   rover_domain_setup.py
	new file:   rover_domain_w_setup.py
	new file:   supervisor.pyx
	new file:   temp_array.hpp
	new file:   temp_array.pxd

zerbeln committed:
Delete mods.py

zerbeln committed:
Removing old files
